features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/RoomDetailsView.kt

```diff
@@ -80,33 +85,50 @@ fun RoomDetailsView(
             .padding(padding)
             .verticalScroll(rememberScrollState())
         ) {
-            HeaderSection(
-                avatarUrl = state.roomAvatarUrl,
-                roomId = state.roomId,
-                roomName = state.roomName,
-                roomAlias = state.roomAlias
-            )
-
-            ShareSection(onShareRoom = onShareRoom)
+            if (state.dmMember == null) {
+                RoomHeaderSection(
+                    avatarUrl = state.roomAvatarUrl,
+                    roomId = state.roomId,
+                    roomName = state.roomName,
+                    roomAlias = state.roomAlias
+                )
+                RoomShareSection(onShareUser = onShareRoom)
+            } else {
+                RoomMemberHeaderSection(
+                    avatarUrl = state.roomAvatarUrl ?: state.dmMember.avatarUrl,
+                    userId = state.dmMember.userId,
+                    userName = state.roomName
+                )
+                RoomMemberShareSection(onShareUser = { onShareMember(state.dmMember) })
```

Comment:
Avoid lambda to avoid recomposition

Reply:
This `dmMember` comes from the presenter, so I need to pass it through the callback lambda somehow. Is there a better way to do this?